docs/agent_hocon_reference.md

```diff
@@ -0,0 +1,497 @@
+# Agent Network HOCON File Reference
+
+This document describes the neuro-san specifications for a single agent network .hocon file
+as used for each [registry](https://github.com/leaf-ai/neuro-san/tree/main/neuro_san/registries)
+in the [neuro-san](https://github.com/leaf-ai/neuro-san) and [neuro-san-demos](https://github.com/leaf-ai/neuro-san-demos)
+repos.
```

Comment:
What you've all been waiting for.
Big new file documenting all the keys supported in the agent hocon files.

Note: I tried doing this in a large table format, but multi-line entries in markdown table cells is just not really a thing. 🤷

Reply:
Last night I came to the conclusion that it would be a decent idea to link to these reference docs in the hocon files themselves.  Will do that in subsequent PR.

Reply:
Good point